debezium-core/src/main/java/io/debezium/util/Threads.java

```diff
@@ -200,10 +211,49 @@ public static Thread timeout(String threadName,
             // Otherwise we've timed out ...
             uponTimeout.run();
         };
-        return new Thread(r, threadName);
+        return new Thread(r, DEBEZIUM_THREAD_NAME_PREFIX + "-timeout-" + threadName);
     }
 
     private Threads() {
     }
 
+    /**
+     * Returns a thread factory that creates threads conforming to Debezium thread naming
+     * pattern {@code debezium-<connector class>-<connector-id>-<thread-name>}.
+     *
+     * @param connector - the source connector class
+     * @param connectorId - the identifier to differentiate between connector instances
+     * @param name - the name of the thread
+     * @param indexed - true if the thread name should be appended with an index
+     * @return the thread factory setting the correct name
+     */
+    public static ThreadFactory threadFactory(Class<? extends SourceConnector> connector, String connectorId, String name, boolean indexed) {
+        LOGGER.info("Requested thread factory for connector {}, id = {} named = {}", connector.getSimpleName(), connectorId, name);
+
+        return new ThreadFactory() {
+            private final AtomicInteger index = new AtomicInteger(0);
+
+            @Override
+            public Thread newThread(Runnable r) {
+                StringBuilder threadName = new StringBuilder(connector.getSimpleName().toLowerCase())
```

Comment:
That's missing the `DEBEZIUM_THREAD_NAME_PREFIX` prefix.

Reply:
Damn, never clean up the code after it is tested :-(